test/Dapr.E2E.Test.Actors.Generators/DaprSidecarFactory.cs

```diff
@@ -0,0 +1,151 @@
+// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Diagnostics;
+
+namespace Dapr.E2E.Test.Actors.Generators;
+
+internal sealed record DaprSidecarOptions(string AppId)
+{
+    public int? AppPort { get; init; }
+
+    public int? DaprGrpcPort { get; init;}
+
+    public int? DaprHttpPort { get; init; }
+
+    public ILoggerFactory? LoggerFactory { get; init; }
+
+    public string? LogLevel { get; init; }
+}
+
+internal sealed class DaprSidecar : IAsyncDisposable
```

Comment:
This is why I asked about the other tests. You don't have to rewrite as much :) However, those have proven to be a bit flaky. In your experience has this approach been working better?